File: mobile/src/main/java/com/alexstyl/gsc/SoundComparer.java
package com.alexstyl.gsc;

import java.util.List;

public class SoundComparer {
    private static final SoundRules sRules = SoundRules.getInstance();

    public SoundComparer() {
    }

    public static boolean areSame(String first, String second) {
        if(first == null) {
            throw new IllegalArgumentException("first");
        } else if(second == null) {
            throw new IllegalArgumentException("second");
        } else {
            return sRules.compare(first, second, false);
        }
    }

    public static boolean startsWith(String first, String second) {
        if(first == null) {
            throw new IllegalArgumentException("first");
        } else if(second == null) {
            throw new IllegalArgumentException("second");
        } else {
            return sRules.compare(first, second, true);
        }
    }

    public static List<Sound> soundsLike(String first) {
        if(first == null) {
            throw new IllegalArgumentException("first");
        } else {
            return sRules.soundsLike(first);
        }
    }
}

File: memento/src/main/java/com/alexstyl/gsc/SoundComparer.kt
package com.alexstyl.gsc

class SoundComparer {
    companion object {
        private val soundRules = SoundRules.INSTANCE

        fun soundTheSame(first: String, second: String) = soundRules.compare(first, second, false)

        fun startsWith(fullWord: String, startsWith: String) = soundRules.compare(fullWord, startsWith, true)
    }
}
